code/iaas/logic/src/main/java/io/cattle/platform/process/containerevent/ContainerEventCreate.java

```diff
@@ -185,9 +182,10 @@ void setHost(ContainerEvent event, Instance instance) {
     void setName(Map<String, Object> inspect, Map<String, Object> data, Instance instance) {
         String name = DataAccessor.fromMap(data).withKey(CONTAINER_EVENT_SYNC_NAME).as(String.class);
         if (StringUtils.isEmpty(name))
-            name = (String)inspect.get(INSPECT_NAME);
+            name = DataAccessor.fromMap(inspect).withKey(INSPECT_NAME).as(String.class);
 
-        name = name.replaceFirst("/", "");
+        if (name != null)
+            name = name.replaceFirst("/", "");
         instance.setName(name);
```

Comment:
Craig, any possible side affects if the name is null, and then we set it on the instance on line 189?